tests/unit/datastore_test.py

```diff
@@ -2,9 +2,195 @@
 
 
 import unittest
-from mock import patch, Mock
-from node.adapter.datastore import DatastoreClient, Rule, Profile, Rules
+from mock import patch, Mock, call
+from node.adapter.datastore import (DatastoreClient,
+                                    Rule,
+                                    Profile,
+                                    Rules,
+                                    Endpoint,
+                                    NoEndpointForContainer)
 from etcd import Client as EtcdClient
+from etcd import EtcdResult
+from netaddr import IPNetwork, IPAddress
+import json
+
+TEST_HOST_PATH = "/calico/host/TEST_HOST"
+IPV4_POOLS_PATH = "/calico/ipam/v4/pool/"
+IPV6_POOLS_PATH = "/calico/ipam/v6/pool/"
+TEST_PROFILE_PATH = "/calico/policy/profile/TEST/"
+ALL_PROFILES_PATH = "/calico/policy/profile/"
+ALL_ENDPOINTS_PATH = "/calico/host/"
+ALL_HOSTS_PATH = "/calico/host/"
+TEST_ENDPOINT_PATH = "/calico/host/TEST_HOST/workload/docker/1234/endpoint/" \
+                     "1234567890ab"
+TEST_CONT_ENDPOINT_PATH = "/calico/host/TEST_HOST/workload/docker/1234/" \
+                          "endpoint/"
+TEST_CONT_PATH = "/calico/host/TEST_HOST/workload/docker/1234/"
+CONFIG_PATH = "/calico/config/"
+
+# 4 endpoints, with 2 TEST profile and 2 UNIT profile.
+EP_56 = Endpoint("567890abcdef", "active", "11-22-33-44-55-66")
+EP_56.profile_id = "TEST"
+EP_78 = Endpoint("7890abcdef12", "active", "11-22-33-44-55-66")
+EP_78.profile_id = "TEST"
+EP_90 = Endpoint("90abcdef1234", "active", "11-22-33-44-55-66")
+EP_90.profile_id = "UNIT"
+EP_12 = Endpoint("1234567890ab", "active", "11-22-33-44-55-66")
+EP_12.profile_id = "UNIT"
+
+
+class TestRule(unittest.TestCase):
+
+    def test_create(self):
+        """
+        Test creating a rule from constructor.
+        """
+        rule1 = Rule(action="allow",
+                     src_tag="TEST",
+                     src_ports=[300, 400])
+        self.assertDictEqual({"action": "allow",
```

Comment:
I  don't think you _need_ to fix this, but nose has bindings to allow more idiomatic (and PEP8-compliant) function calls, e.g. assert_dict_equal, setup, teardown, etc.

Not well documented, but I'm pretty sure it's there.